### PR TITLE
mrc-3559 Display comparison plot error message

### PR DESCRIPTION
**Pull request by Nicolas-Dolan**
## Description

Separates comparison plot errors from calibrate errors in the store and displays them in the model output.

## Type of version change
Minor

## Checklist

- [x] I have incremented version number, or version needs no increment
- [x] The build passed successfully, or failed because of ADR tests

### CODE REVIEW

File: src/app/static/src/app/components/modelOutput/ModelOutput.vue [hunk 1]
```diff
@@ -100,6 +100,7 @@
                     :formatFunction="formatBarchartValue"
                     :showRangesInTooltips="true"
                     @update="updateComparisonPlotSelectionsAndXAxisOrder"></bar-chart-with-filters>
+                <error-alert v-if="hasComparisonPlotError" :error="comparisonPlotError"></error-alert>
```
Comment (LekanAnni) [inline src/app/static/src/app/components/modelOutput/ModelOutput.vue:103]:
```suggestion
                <error-alert v-if="!!comparisonPlotError" :error="comparisonPlotError"></error-alert>
```

File: src/app/static/src/app/components/modelOutput/ModelOutput.vue [hunk 2]
```diff
@@ -236,6 +240,10 @@
                     featureLevels: state => state.shape!.filters.level_labels || []
                 }
             ),
+            ...mapStateProps<ModelCalibrateState, keyof Computed>("modelCalibrate", {
+                comparisonPlotError: state => state.comparisonPlotError,
+                hasComparisonPlotError: state => !!state.comparisonPlotError,
```
Comment (LekanAnni) [inline src/app/static/src/app/components/modelOutput/ModelOutput.vue:245]:
```suggestion

```

File: src/app/static/src/app/components/modelOutput/ModelOutput.vue [hunk 3]
```diff
@@ -192,6 +194,8 @@
         filteredBubblePlotIndicators: ChoroplethIndicatorMetadata[],
         barchartFlattenedXAxisFilterOptionIds: string[]
         comparisonPlotFlattenedXAxisFilterOptionIds: string[]
+        comparisonPlotError: Error | null
+        hasComparisonPlotError: boolean
```
Comment (LekanAnni) [inline src/app/static/src/app/components/modelOutput/ModelOutput.vue:198]:
```suggestion

```

**Review by LekanAnni — CHANGES_REQUESTED** [review]
Looks great, I can't see, to find comparison plot, is it behind future switch ?

One or two suggestions below.

**Review by EmmaLRussell — APPROVED** [review]
LGTM

File: src/app/static/src/app/components/modelOutput/ModelOutput.vue (hunk 1 above)
Comment (Nicolas-Dolan) [reply src/app/static/src/app/components/modelOutput/ModelOutput.vue:103]: I can change this, but this is the same pattern used in `modelOptions.vue`

**Comment by Nicolas-Dolan** [comment]
> Looks great, I can't see, to find comparison plot, is it behind feature switch ?

To view go to the Review Output tab and visit http://localhost:8080/?comparisonPlot=1

**Review by LekanAnni — APPROVED** [review]
LGTM 👍 

Thanks

**Pull request merged**